### ISSUE TITLE
SyntaxError instead of JsonWebTokenError

### ISSUE THREAD

**Issue by sm2017** [auth0/node-jsonwebtoken#652]
### Description

Sometime verify throws `SyntaxError` instread of `JsonWebTokenError`

### Reproduction

```js
var jwt = require("jsonwebtoken")
const malformed_token='eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c';

jwt.verify(
        malformed_token,
        'Secret',
        {
          algorithms: [ 'HS256' ],
        }
      );
```

### Environment
**Version of this library used:** 8.5.1

**Comment by sm2017** [comment]
@ziluvatar reply please

**Comment by ziluvatar** [comment]
The payload of that token is not well built, if you try to base64 decode you'll find that it does not contain a JSON payload, you will see something like

```
{"sub":"1234567890","name":"HX]MLML
```

**Comment by sm2017** [comment]
@ziluvatar yes , I know, I just tild that throwing `SyntaxError`  is not good behavior, instead we must throw `JsonWebTokenError`

Do you have other idea?

**Comment by ziluvatar** [comment]
I tagged this and another similar one as `bug`. I mean, you are right we should raise one of our errors.

**Comment by sm2017** [comment]
@ziluvatar The issue a about stale , Do you have plan to fix it? and the issue tagged as a `bug`

**Comment by sm2017** [comment]
@ziluvatar 😓

**Comment by asvsfs** [comment]
I've changed a single line returning default error to JsonWebTokenError type , see if this solves the problem #735
